src/sentry/static/sentry/app/views/settings/components/forms/selectField.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import {OptionsType, ValueType} from 'react-select';
+
+import SelectControl, {ControlProps} from 'app/components/forms/selectControl';
+import {Choices, SelectValue} from 'app/types';
+import InputField from 'app/views/settings/components/forms/inputField';
+
+type InputFieldProps = React.ComponentProps<typeof InputField>;
+
+type Props<OptionType> = InputFieldProps &
+  Omit<ControlProps<OptionType>, 'onChange'> & {
+    /**
+     * Should the select be clearable?
+     */
+    allowClear?: boolean;
+    /**
+     * Should the select allow empty values.
```

Comment:
should we add `?` here?
```suggestion
     * Should the select allow empty values?
```